sdl_android/src/main/java/com/smartdevicelink/protocol/WiProProtocol.java

```diff
@@ -468,6 +473,20 @@ else if (frameInfo == FrameDataControlFrameType.StartSession.getValue()) {
 							//At this point we have confirmed the negotiated version between the module and the proxy
 							protocolVersion = new Version((String)version);
 						}
+					}else if(serviceType.equals(SessionType.NAV)){
+						SdlSession session = sdlconn.findSessionById((byte) packet.sessionId);
+						ImageResolution acceptedResolution = new ImageResolution();

```

Comment:
All of this can be moved under the `if(session!=null)` check because if it's null there's no point in performing any of this logic.